src/pip/_internal/commands/inspect.py

```diff
@@ -0,0 +1,91 @@
+import json
+import sys
+from optparse import Values
+from typing import Any, Dict, List
+
+from pip._vendor.packaging.markers import default_environment
+
+from pip import __version__
+from pip._internal.cli import cmdoptions
+from pip._internal.cli.req_command import Command
+from pip._internal.cli.status_codes import SUCCESS
+from pip._internal.metadata import BaseDistribution, get_environment
+from pip._internal.utils.compat import stdlib_pkgs
+from pip._internal.utils.urls import path_to_url
+
+
+class InspectCommand(Command):
+    """
+    Inspect the content of a Python environment.
+    """
+
+    ignore_require_venv = True
+    usage = """
+      %prog [options]"""
+
+    def add_options(self) -> None:
+        self.cmd_opts.add_option(
+            "-l",
+            "--local",
+            action="store_true",
+            default=False,
+            help=(
+                "If in a virtualenv that has global access, do not list "
+                "globally-installed packages."
+            ),
+        )
+        self.cmd_opts.add_option(
+            "--user",
+            dest="user",
+            action="store_true",
+            default=False,
+            help="Only output packages installed in user-site.",
+        )
+        self.cmd_opts.add_option(cmdoptions.list_path())
+        self.parser.insert_option_group(0, self.cmd_opts)
+
+    def run(self, options: Values, args: List[str]) -> int:
+        cmdoptions.check_list_path_option(options)
+        dists = get_environment(options.path).iter_installed_distributions(
+            local_only=options.local,
+            user_only=options.user,
+            skip=set(stdlib_pkgs),
+        )
+        output = {
+            "version": "0",
+            "pip_version": __version__,
+            "installed": [self._dist_to_dict(dist) for dist in dists],
+            "environment": default_environment(),
+            # TODO tags? scheme?
+        }
+        json.dump(output, sys.stdout, indent=2, ensure_ascii=True)
```

Comment:
I wonder if we should do `rich.print(rich.json.JSON(output))` instead.

Reply:
Hm, the default rich pager is not working well on my system, so I'm not gonna take risks with that. Rich JSON works fine so I included that.

Reply:
@pradyunsg unfortunately `rich.print(rich.json.JSON.from_data(output))` is inserting newlines in long text fields (such as `metadata.description`), resulting in invalid JSON even when redirected to a file. So I'm going to stick to `json.dump` for now. Piping it to `jq` gives colored output similar to what rich does.

Reply:
It should be possible to disable wrapping with rich -- I did that for our logging output. I'm pretty sure there's a flag on rich.print for that.

Otherwise, try `logging.info("[display-rich] %s", JSON(data))`?

Reply:
Ah, alright, rich has `print_json` that does the right thing.